packages/core/types/src/modules/entity-service/params/sort.ts

```diff
@@ -75,16 +75,24 @@ export type ArrayNotation<TSchemaUID extends Common.UID.Schema> = Any<TSchemaUID
  * type F = 'title'; // ❌
  */
 export type ObjectNotation<TSchemaUID extends Common.UID.Schema> = {
-  // First level sort
-  [key in SingleAttribute<TSchemaUID>]?: OrderKind.Any;
-} & {
-  // Deep sort, only add populatable keys that have a
-  // target (remove dynamic zones and other polymorphic links)
-  [key in Attribute.GetKeysWithTarget<TSchemaUID>]?: ObjectNotation<
-    Attribute.GetTarget<TSchemaUID, key>
-  >;
+  [TKey in ObjectNotationKeys<TSchemaUID>]?: TKey extends SingleAttribute<TSchemaUID>
+    ? // First level sort (scalar attributes, id, ...)
+      OrderKind.Any
+    : // Deep sort (relations with a target, components, media, ...)
+      ObjectNotation<Attribute.GetTarget<TSchemaUID, TKey>>;
 };
 
+/**
+ * Represents the keys of an object notation for a sort
+ * - SingleAttribute<TSchemaUID> represents a union of every non-populatable attribute based on the passed schema UID
+ * - Attribute.GetKeysWithTarget<TSchemaUID> provides keys with a target from the passed schema UID.
+ *
+ * This means that every member of ObjectNotationKeys can represent either a single non-populatable attribute or an attribute with a target.
+ */
+type ObjectNotationKeys<TSchemaUID extends Common.UID.Schema> =
+  | SingleAttribute<TSchemaUID>
```

Comment:
you don't have to change it but that type sounds exotic. Shouldn't it be in some Attribute.GetXXXXKeys util ?

Reply:
This is only related to the sort object notation though. I don't think it makes sense to expose it elsewhere (especially from attribute utils)